internal/runtimehandlerhooks/high_performance_hooks_linux.go

```diff
@@ -1233,6 +1270,10 @@ func isCgroupParentBestEffort(s *sandbox.Sandbox) bool {
 }
 
 func isContainerRequestWholeCPU(cSpec *specs.Spec) bool {
+	if isContainerCPUEmpty(cSpec) || cSpec.Linux.Resources.CPU.Shares == nil {
+		return false
+	}
+
```

Comment:
This doesn't seem related to your change - please explain why this is needed?

Reply:
Because since I introduced my changes, there was duplication in the code in various locations of:

		spec.Linux.Resources == nil ||
		spec.Linux.Resources.CPU == nil ||
		spec.Linux.Resources.CPU.Cpus == ""

I'll branch it off into a different, small commit instead

Reply:
OK - in this function it appears we are expecting cSpec.Linux.Resources.CPU.Shares to be valid - would it be a good idea to insert a "log.Error" here before returning false?

Reply:
It may be because our code path always put the value there by coincidence.